File: DataStructure/36. Valid Sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = [set() for i in range(9)]
        cols = [set() for i in range(9)]
        boxs = [set() for i in range(9)]
        
        for r in range(len(board)):
            for c in range(len(board[r])):
                s = board[r][c]
                if s != '.':
                    if s in rows[r] or s in cols[c] or s in boxs[r // 3 * 3 + c // 3]:
                        return False
                    rows[r].add(s)
                    cols[c].add(s)
                    boxs[r//3*3+c//3].add(s)
                    print(r//3*3+c//3)
        
        print(boxs)
        return True

File: DataStructure/36. Valid Sudoku.cs
public class Solution {
    public bool IsValidSudoku(char[][] board) {
        List<HashSet<int>> rows = new List<HashSet<int>>();
        List<HashSet<int>> cols = new List<HashSet<int>>();
        List<HashSet<int>> boxes = new List<HashSet<int>>();
        
        for (int i = 0; i < board.GetLength(0); i++)
        {
            rows.Add(new HashSet<int>());
            cols.Add(new HashSet<int>());
            boxes.Add(new HashSet<int>());
        }
        
        for (int r = 0; r < rows.Count(); r++)
        {
            for (int c = 0; c < cols.Count(); c++)
            {
                char s = board[r][c];
                if (s != '.')
                {
                    if (rows[r].Contains(s) || cols[c].Contains(s) || boxes[r / 3 * 3 + c / 3].Contains(s))
                    {
                        return false;
                    }
                    rows[r].Add(s);
                    cols[c].Add(s);
                    boxes[r / 3 * 3 + c / 3].Add(s);
                }
            }
        }        
        return true;
    }
}